pkgs/apps/ambertools/default.nix

```diff
@@ -0,0 +1,143 @@
+{ buildPythonPackage
+, lib
+, requireFile
+, makeWrapper
+  # Python dependencies
+, numpy
+, scipy
+, matplotlib
+, setuptools
+  # Native dependencies
+, gfortran
+, cmake
+, readline
+, perl
+, flex
+, bison
+, zlib
+, boost
+, netcdf
+, netcdffortran
+, fftw
+, blas
+, lapack
+, protobuf
+, plumed
+, apbs
+, arpack
+, runtimeShell
+}:
+
+buildPythonPackage rec {
+  pname = "AmberTools";
+  version = "24";
+
+  src = requireFile {
+    name = "AmberTools${version}.tar.bz2";
+    sha256 = "sha256-UvtPszcKibfOc4otw+UTwvwZQ/3ktDgYRtnnXMSNhA8=";
+    url = "https://ambermd.org/AmberTools.php";
+  };
+
+  nativeBuildInputs = [
+    cmake
+    gfortran
+    flex
+    bison
+    makeWrapper
+  ];
+
+  buildInputs = [
+    zlib
+    boost
+    blas
+    lapack
+    netcdffortran
+    fftw
+    protobuf
+    plumed
+    arpack
+    apbs
+    readline
+  ];
+
+  format = "other";
+
+  cmakeFlags = [
+    "-DCOMPILER=AUTO"
+    "-DDOWNLOAD_MINICONDA=OFF"
+    "-DOPENMP=ON"
+    "-DTRUST_SYSTEM_LIBS=ON"
+  ];
+
+  propagatedBuildInputs = [
+    perl
+  ];
+
+  dependencies = [
+    numpy
+    scipy
+    matplotlib
+    setuptools
+  ];
+
+  buildPhase = ''
+    runHook preBuild
+
+    make -j $NIX_BUILD_CORES
+
+    runHook postBuild
+  '';
+
+  installPhase =
+    let
+
+      wrongBash = [
+        "am1bcc"
+        "antechamber"
+        "atomtype"
+        "bondtype"
+        "espgen"
+        "match"
+        "match_atomname"
+        "parmcal"
+        "parmchk2"
+        "prepgen"
+        "reduce"
+        "residuegen"
+        "respgen"
+        "XrayPrep"
+      ];
+    in
+    ''
+      runHook preInstall
+
+      make install
+
+      # Some scripts hardcode /bin/bash. Not only necessary as their shebang, but
+      # some also generate bash scripts with wrong shebangs.
+      for PROG in ${builtins.toString wrongBash}; do
+        substituteInPlace $out/bin/$PROG --replace-fail '#!/bin/bash' '#!${runtimeShell}'
+      done
+
+      substituteInPlace $out/amber-interactive.sh --replace-fail '#! /bin/bash' '#!${runtimeShell}'
+
+      # Avoids sourcing amber.sh before running ambertools by setting the required
+      # variables via wrappers for each program.
+      for PROG in $out/bin/*; do
+        if [[ -f $PROG ]]; then
+          wrapProgram $PROG \
+            --set AMBERHOME $out \
+            --set QUICK_BASIS=$out/AmberTools/src/quick/basis
+        fi
+      done
+
+      runHook postInstall
+    '';
+
+  meta = with lib; {
+    description = "Tools for molecular mechanics and molecular dynamics with AMBER";
+    homepage = "https://ambermd.org/AmberTools.php";
+    license = with licenses; [ lgpl3 bsd3 mit asl20 gpl3Only gpl2Only ];
```

Comment:
E.g.: "You may install the Software on computers at your site, or at other locations, provided that reasonable steps are taken to restrict use of the software to you and members of your organization. You may not distribute copies of the Software to others outside of your organization, and may make only those copies of the Software which are necessary to install and use it as permitted by this Agreement, or are for purposes of backup and archival records."

We can not distribute it via cachix.

Reply:
No I think this part applies only to Amber itself, not the AmberTools. For AmberTools the website states
" AmberTools is a free, useful standalone package and a prerequisite for installing Amber itself. The AmberTools suite is free of charge, and its components are mostly released under the GNU General Public License (GPL). A few components are included that are in the public domain or which have other, open-source, licenses. The libsander and libpbsa libraries use the LGPL license. "

Reply:
There is another license file in the tarball that lists the licenses for all components and none of that is unfree. Their conda channel also only lists free licenses: https://anaconda.org/conda-forge/ambertools

Reply:
ok, I see. However, the fact the source code it is hidden behind a "signup form" makes it unfree from at least the technical perspective (i.e. we can not provide a direct download link). The public hydra gets stuck trying to get the sources.

There seems to be a direct download link for the [conda forge build sources](https://github.com/conda-forge/ambertools-feedstock/blob/main/recipe/meta.yaml), but 24 is not downloadable via that that URL.
Maybe write an email to the developers and ask them to put it there and if they are okay with us distributing the source code/build products?

Reply:
Yes, the Conda channel lacks behind and does so differently for different architectures. Idk why. Anyway, I wrote a message to the e-mail address provided for license questions, so let's see :smile:

Reply:
> If we are allowed to distribute the sources, it should be possible to manually add them once to the nix store via nix store add-file. 

Yes, that is possible, but I would I try to avoid creating such technical baggage. 

> It's manual and a little bit ugly but would allow the Hydra to build and distribute. For Cachix we would need to push manually from a local machine, I suppose?

The public hydra just pushes everything it builds to the cachix. If it can build it the build product ends up on cachix. The sources, would have to be pushed manually though (that is then already two manual steps)

Reply:
> If we are allowed to distribute the sources, it should be possible to manually add them once to the nix store via nix store add-file. 

Yes, that is possible, but I would I try to avoid creating such technical baggage. 

> It's manual and a little bit ugly but would allow the Hydra to build and distribute. For Cachix we would need to push manually from a local machine, I suppose?

The public hydra just pushes everything it builds to the cachix. If it can build it the build product ends up on cachix. The sources, would have to be pushed manually.

Reply:
Hm, no answer from the Amber developers yet and technical problems with caching not really solved. Short term suggestion: let's add `lib.meta.dontDistribute` for the moment so that we have a working version merged. If we get an answer, we can just remove this line. What do you think?

Reply:
That sounds like a good idea.

Reply:
Alright, pushed a commit with `meta.hydraPlatforms = [ ]` added, which is exactly what `lib.meta.dontDistribute` does.